sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,389 @@
+# Troubleshooting Azure Monitor Query client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Monitor Query client library for JavaScript.
+
+## Table of contents
+
+- [General troubleshooting](#general-troubleshooting)
+  - [Enable client logging](#enable-client-logging)
+  - [Troubleshooting authentication issues with logs and metrics query requests](#authentication-errors)
+- [Troubleshooting logs query](#troubleshooting-logs-query)
+  - [Troubleshooting insufficient access error](#troubleshooting-insufficient-access-error-for-logs-query)
+  - [Troubleshooting invalid Kusto query](#troubleshooting-invalid-kusto-query)
+  - [Troubleshooting empty log query results](#troubleshooting-empty-log-query-results)
+  - [Troubleshooting server timeouts when executing logs query request](#troubleshooting-server-timeouts-when-executing-logs-query-request)
+  - [Troubleshooting partially successful logs query requests](#troubleshooting-partially-successful-logs-query-requests)
+- [Troubleshooting metrics query](#troubleshooting-metrics-query)
+  - [Troubleshooting authorization failed error](#troubleshooting-authorization-failed-error-for-metrics-query)
+  - [Troubleshooting unsupported granularity](#troubleshooting-unsupported-granularity-for-metrics-query)
+
+## General troubleshooting
+
+### Enable client logging
+
+To troubleshoot issues with Azure Monitor Query library, it's important to first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+The Azure client libraries for JavaScript allow you to enable logging either through the environment variable or at runtime.
+
+The following log levels are supported. They're listed from most to least verbose.
+
+- verbose
+- info
+- warning
+- error
+
+When setting a log level, either programmatically or via the `AZURE_LOG_LEVEL` environment variable, any logs that are written using a log level equal to or less than the one you choose will be emitted. For example, setting the log level to `warning` will cause all logs that have the log level `warning` or `error` to be emitted.
+
+#### Logging via environment variable
+
+To see a log of HTTP requests and responses, set the `AZURE_LOG_LEVEL` environment variable to `info`.
+
+Add the following to the .env
+```text
+AZURE_LOG_LEVEL = info
+```
+
+Add the following to the code 
+```ts
+import * as dotenv from "dotenv";
+dotenv.config({ path: ".env" });
+```
+
+#### Logging using setLogLevel
+
+Alternatively, logging can be enabled at runtime by calling `setLogLevel` in the `@azure/logger`:
+
+```ts
+import { setLogLevel } from "@azure/logger";
+
+setLogLevel("info");
+```
+
+**NOTE**: When logging the body of request and response, ensure that they don't contain confidential information.
```

Comment:
maybe in this note we should mention we already sanitize headers like 'Authorization' that contain secrets, and a link to the list of which headers are actually logged?

Reply:
@xirzec Do you know where is the list of headers that are logged listed? I couldn't find them under logger readme - https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/core/logger Also I was thinking if i should update the logger readme with that info instead?